test/e2e/framework/pods.go

```diff
@@ -313,3 +316,54 @@ func (c *PodClient) PodIsReady(name string) bool {
 	ExpectNoError(err)
 	return podutil.IsPodReady(pod)
 }
+
+// ExecWithOptions executes a command in the specified container,
+// returning stdout, stderr and error. `options` allowed for
+// additional parameters to be passed.
+func (c *PodClient) ExecWithOptions(options ExecOptions, cs clientset.Interface) (string, string, error) {
```

Comment:
I have two questions:

1)  why we need two new methods on PodClient? These two methods do not access PodClient fields. And we already have [exec methods on Framework](https://github.com/kubernetes/kubernetes/blob/ed5b03831385b0071b127ff6c329ae344b9780f6/test/e2e/framework/exec_util.go#L50), I see the usage of `PodClient()` [here](https://github.com/kubernetes/kubernetes/pull/84042/files#diff-0d21a89b3adc375caf1b557b014e1111R68), It still use the framework ClientSet. 

2) If we need to, I think it's better to have some subroutines and reuse code.

Reply:
I might have suggested that to @okartau as one option before he started working on the code. But as you noticed, in the end the resulting methods are not related to `PodClient`, which indeed indicates that they belong elsewhere.

But before we discuss where a better place for them might be, do we actually still need this PR? It was supposed to solve https://github.com/kubernetes/kubernetes/issues/83913, but as you pointed out in https://github.com/kubernetes/kubernetes/issues/83913#issuecomment-548683208, that problem was already solved (differently, but still solved) in https://github.com/kubernetes/kubernetes/pull/84444, correct?

In that case we can simply close this PR unless someone sees some value in these enhancements.

Reply:
I am ok with closing.

Reply:
HostExec is an alternative way to `ssh.SSH`. To avoid using kubectl, `*Framework` is still needed to access ExecWithOptions/ExecCommandInContainerWithFullOutput methods.

Reply:
@cofyc: so you are saying that https://github.com/kubernetes/kubernetes/pull/84444 does not remove the kubectl dependency completely and therefore this PR is still needed? That part is not clear to me.

But we have the source, so let's use that... :smile:

In current Kubernetes master, https://github.com/kubernetes/kubernetes/blob/08410cbf06f1c8649157407c660b1c5cb528947b/test/e2e/storage/testsuites/multivolume.go#L368 uses this chain of calls:
- https://github.com/kubernetes/kubernetes/blob/08410cbf06f1c8649157407c660b1c5cb528947b/test/e2e/storage/utils/utils.go#L567
- https://github.com/kubernetes/kubernetes/blob/08410cbf06f1c8649157407c660b1c5cb528947b/test/e2e/storage/utils/utils.go#L71
- https://github.com/kubernetes/kubernetes/blob/08410cbf06f1c8649157407c660b1c5cb528947b/test/e2e/storage/utils/utils.go#L66
- https://github.com/kubernetes/kubernetes/blob/08410cbf06f1c8649157407c660b1c5cb528947b/test/e2e/framework/util.go#L1109
- -> kubectl still needed

@okartau: I'm not sure why these new functions were necessary. Can you use the existing ExecWithOptions/ExecCommandInContainerWithFullOutput from https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/exec_util.go?

Reply:
yes, this PR is still needed.

`(*Framework).ExecWithOptions/ExecCommandInContainerWithFullOutput` are used to exec command in an existing pod. However, `HostExec` is used to run commands on the host (by using `nsenter` to entering into host namespace). They are for different purposes.